src/Remote/Importer.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Castor\Remote;
+
+use Castor\GlobalHelper;
+use Castor\PathHelper;
+use Castor\Remote\Exception\ImportError;
+use Castor\Remote\Exception\InvalidImportFormat;
+use Psr\Log\LoggerInterface;
+
+use function Castor\import;
+
+/** @internal */
+class Importer
+{
+    public function __construct(
+        private readonly LoggerInterface $logger,
+        private readonly Composer $composer,
+        /** @var array<string, array<string, array<string|null>>> */
+        private array $imports = [],
+    ) {
+    }
+
+    /**
+     * @param ?array{
+     *     url?: string,
+     *     type?: "git" | "svn",
+     *     reference?: string,
+     * } $source
+     */
+    public function importFunctionsFrom(string $scheme, string $package, ?string $file = null, ?string $version = null, ?string $vcs = null, ?array $source = null): void
+    {
+        if (!$this->allowsRemote()) {
+            throw new ImportError(sprintf('Remote imports are disabled, skipping import of "%s".', $package));
+        }
+
+        if (isset($this->imports[$package]) && ($requiredVersion = array_key_first($this->imports[$package])) !== $version) {
+            throw new ImportError(sprintf('The package "%s" is already required in version "%s", could not require it in version "%s"', $package, $requiredVersion, $version));
+        }
+
+        if (!preg_match('#^(?<organization>[^/]+)/(?<repository>[^/]+)$#', $package)) {
+            throw new InvalidImportFormat(sprintf('The import path must be formatted like this: "%s://<organization>/<repository>".', $scheme));
+        }
+
+        if ('composer' === $scheme) {
+            if (null !== $source) {
+                throw new InvalidImportFormat('The "source" argument is not supported for Composer/Packagist packages.');
+            }
+
+            $this->importPackageWithComposer($package, version: $version ?? '*', repositoryUrl: $vcs, file: $file);
+
+            return;
+        }
+
+        if ('package' === $scheme) {
+            if (null !== $version || null !== $vcs) {
+                throw new InvalidImportFormat('The "source" and "vcs" arguments are not supported for non-Composer packages.');
+            }
+            if (null === $source) {
+                throw new InvalidImportFormat('The "source" argument is required for non-Composer packages.');
+            }
+
+            $this->importPackageWithComposer($package, version: 'v1', source: $source, file: $file);
+
+            return;
+        }
+
+        throw new InvalidImportFormat(sprintf('The import scheme "%s" is not supported.', $scheme));
```

Comment:
In all previous exception, could you add more informations, 

* the files where it's defined (if you can, otherwise, I'll do), 
* the name of the package, etc

Reply:
Same, will be done higher